core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1144,13 +1144,12 @@ class DAGScheduler(
         null
       }
 
-    // The success case is dealt with separately below.
-    // TODO: Why post it only for failed tasks in cancelled stages? Clarify semantics here.
-    if (event.reason != Success) {
-      val attemptId = task.stageAttemptId
-      listenerBus.post(SparkListenerTaskEnd(
-        stageId, attemptId, taskType, event.reason, event.taskInfo, taskMetrics))
-    }
+    // Note: this stage may already have been canceled, in which case this task end event
+    // maybe posted after the stage completed event. There's not much we can do here without
+    // introducing additional complexity in the scheduler to wait for all the task end events
+    // before posting the stage completed event.
```

Comment:
So it doesn't seem like we need to be so specific here.  I don't see the benefit of adding the spark jira, you can go back to history to find change if needed I don't see that the information there provides much benefit over the rest of the comment.  

I'll update shortly, let me know what you think.